src/components/calcite-filter/calcite-filter.tsx

```diff
@@ -0,0 +1,103 @@
+import { Component, Element, Event, EventEmitter, Host, Prop, h } from "@stencil/core";
+import { debounce, forIn } from "lodash-es";
+
+const filterDebounceInMs = 250;
+
+@Component({
+  tag: "calcite-filter",
+  styleUrl: "calcite-filter.scss",
+  shadow: true
+})
+export class CalciteFilter {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * The input data. The filter uses this as the starting point, and returns items
+   * that contain the string entered in the input, using a partial match and recursive search.
+   */
+  @Prop() data: object[];
+
+  /**
+   * A text label that will appear next to the input field.
+   */
+  @Prop() textLabel: string;
+
+  /**
+   * Placeholder text for the input element's placeholder attribute
+   */
+  @Prop() textPlaceholder: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteFilterElement;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calciteFilterChange: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  filter = debounce((value: string): void => {
+    const regex = new RegExp(value, "ig");
+    const find = (input, regex) => {
+      let found = false;
+      forIn(input, (value) => {
+        if (typeof value === "function") {
+          return;
+        }
+        if (Array.isArray(value) || (typeof value === "object" && value !== null)) {
+          if (find(value, regex)) {
+            found = true;
+          }
+        } else if (regex.test(value)) {
+          found = true;
+        }
+      });
+      return found;
+    };
+
+    const result = this.data.filter((item) => {
+      return find(item, regex);
+    });
+
+    this.calciteFilterChange.emit(result);
+  }, filterDebounceInMs);
+
+  inputHandler = (event: Event): void => {
+    const target = event.target as HTMLInputElement;
+    this.filter(target.value);
+  };
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  render() {
+    return (
+      <Host>
+        <label>
```

Comment:
Boulder:

From aXe:

```
Form elements must have labels

Issue description
Ensures every form element has a label

Impact: critical
Learn more
Element location
#one calcite-filter input
Element source
<input placeholder="Filter results">
```

Reply:
that doesn't make sense - there is a label.

Reply:
@pr3tori4n probably because `textLabel` is empty by default and not being set on the component. Maybe it should have a default value if not set??